### ISSUE TITLE
Deprecating and removing support for ghc 7.6

### ISSUE THREAD

**Issue by lennart** [tidalcycles/Tidal#209]
Since the latest version of tidal now includes the `OverloadedLists` extension, we cannot easily support ghc 7.6 anymore (it simply does not have this extension).

due to constrained maintenance resources I'd vote for dropping support for it and therefore removing ghc 7.6 from the test suite.

side note: the "travis badge" should be green in case tidal compiles on all supported platforms (note however, that we have to continually update the readme to point to the correct branch it is on)

**Comment by lennart** [comment]
@yaxu @vivid-synth thoughts on this?

**Comment by yaxu** [comment]
I'm a bit unsure about OverloadedLists, in a way it's convenient, but confusing, because it makes patterns that look like lists but aren't. Currently it's not switched on for anyone - this needs to be done in the editor plugins. I did have a bit of a look around to see if there's some way of doing conditional compilation so we could only enable it for compilers that support it but didn't immediately find that.
Basically I'm not sure whether OverloadedLists is worth it the hassle for people with ancient ghc installs or not. We could potentially move this to its own package, or just give it as a code recipe for people to run.

**Comment by telephon** [comment]
It seems at least a little incoherent to allow `sound ["bd", "sn"]` but not `sound (["bd", "sn"] ++ ["arpy", "cp"])`. Why is it convenient?

**Comment by yaxu** [comment]
Just the convenience of not giving `listToPat` explicitly, just as OverloadedStrings means we don't have to explicitly specify the parser `p` whenever a string is turned into a pattern.

**Comment by yaxu** [comment]
I removed this list overloading for now and the travis build is working again.

**Comment by telephon** [comment]
ok, I see. If you can't treat lists as lists, you don't really benefit from this convenience, as it seems.

**Comment by vivid-synth** [comment]
Typically for Haskell "good compiler support" is something like 3 versions, so 7.8, 7.10, and 8.0 (and soon 8.2!) should be plenty. I'm similarly mixed about OverloadedLists, but I think you'd be "within your rights" to remove 7.6 support if you needed to, for this or other reasons.

Timeline of ghc releases is here: https://www.haskell.org/ghc/
